pkg/build/api/validation/validation_test.go

```diff
@@ -105,8 +105,18 @@ func TestBuildConfigValidationFailure(t *testing.T) {
 				DockerImageReference: "repository/data",
 			},
 		},
+		Triggers: []buildapi.BuildTriggerPolicy{
+			{
+				Type:        buildapi.ImageChangeBuildTriggerType,
+				ImageChange: &buildapi.ImageChangeTrigger{},
+			},
+			{
+				Type:        buildapi.ImageChangeBuildTriggerType,
+				ImageChange: &buildapi.ImageChangeTrigger{},
+			},
+		},
 	}
-	if result := ValidateBuildConfig(buildConfig); len(result) != 1 {
+	if result := ValidateBuildConfig(buildConfig); len(result) != 2 {
```

Comment:
I might be wrong, but this test triggers my smell alarm.

We're just testing that `result`, which is of type `fielderrors.ValidationErrorList`, has length 2... there could be any validation errors in the BuildConfig, and the test doesn't document nor check which errors are that. It could fail or succeed for different reasons.

And we could have a more [useful failure message](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures).

I could patch this as part of my work on #2594 once I get it ready.